### PR TITLE
New Datepicker

### LINKED ISSUE #2379: Alternative datepicker
### Description

As we use a deprecated version of material-UI and wish to get away from material-UI we should find alternative datepicker to handle datepicking in app-frontend. 

Alternative: https://daypicker.dev/ 

### Additional Information

Should correspond with change of moment - date/time tools (#191)

### Tasks

- [ ] Replace with new solution
- [ ] Correspond with design sketches
- [ ] Regression tests

### Acceptance Criterias

A viable alternative is identified and has replaced material-UI datepicker where used. Should cover all functionality that exists in the current solution.

### PR DESCRIPTION
**Pull request by paal2707**
## Description

Refactor datepicker component to get rid of MUI datepicker and momentjs, using React-daypicker instead.

## Related Issue(s)

- closes #2379 

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [x] Unit test(s) have been added/updated
  - [ ] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [x] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [ ] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Comment by paal2707** [comment]
Needs to:
- Fix jest test for render on small screen
- Append format to label text instead of placeholder on input. Ask @Edavda or @olavflar
- Translate aria-label for comboboxes in calendar
- Translate aria-label for selected day in calendar
- Cypress tests

File: src/utils/dateHelpers.ts [hunk 1]
```diff
@@ -1,58 +1,145 @@
...
+  if (format === 'DD.MM.YYYY') {
+    return 'dd.MM.y';
+  }
+  if (format === 'DD/MM/YYYY') {
+    return 'dd/MM/yyyy';
+  }
+  if (format === 'YYYY-MM-DD') {
+    return 'yyyy-MM-dd';
+  }
+  return format;
```
[the remark this answers is not in the document]
Comment (paal2707) [reply src/utils/dateHelpers.ts:40]: no, good point. Trying to check it now.

[the remark this answers is not in the document]
Comment (paal2707) [reply src/utils/dateHelpers.ts:40]: Have checked it out now and i cannot find any othe formats

File: src/layout/Datepicker/DropdownCaption.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  const fromDate = subYears(calendarMonth.date, 100);
+  const toDate = addYears(calendarMonth.date, 100);
+
+  const years = getYears(fromDate, toDate, calendarMonth.date.getFullYear()).reverse();
+  const months = getMonths(fromDate, toDate, calendarMonth.date);
+  const yearDropdownLabel = langAsString('date_picker.aria_label_year_dropdown');
+  const MonthDropdownLabel = langAsString('date_picker.aria_label_month_dropdown');
+  return (
+    <>
+      <div style={{ display: 'flex', marginBottom: '10px' }}>
```
Comment (cammiida) [inline src/layout/Datepicker/DropdownCaption.tsx:42]: Vet ikke hva vi tenker mht `styles` vs `CSS Modules` her, men burde kanskje holde oss så mye som mulig til det ene eller andre? 🤷‍♀️  dunno

File: src/utils/dateHelpers.ts [hunk 3]
```diff
@@ -1,58 +1,145 @@
...
-  return (
-    (timestamp === false
-      ? date?.format(DatepickerSaveFormatNoTimestamp)
-      : date?.format(DatepickerSaveFormatTimestamp)) ?? ''
-  );
+/**
+ * The datepicker component does not support standard formats for input so the format we pass to the datepicker need to be massaged a bit.
+ * @deprecated
+ */
+export function convertToDatepickerFormat(format: string): string {
```
Comment (cammiida) [inline src/utils/dateHelpers.ts:54]: Har du laget en ny funksjon som du har markert som `deprecated`? 😂  Eller ser jeg feil? Kan denne byttes ut med feks `convertLegacyFormat` eller noe?

**Review by cammiida — APPROVED** [review]
Sykt bra jobba. Funker veldig bra og enkelt å følge koden 👏 🚀 

Har bare super småpirk og så lurer jeg på om man kan gjøre inputen litt mindre så den matcher de andre? 😊 Sånn som i skjermbildet her
image

**Comment by cammiida** [comment]
Kanskje du kan fikse teksten som er referert til her i samme slengen? 🤩 https://digdir.slack.com/archives/C0760NPT2BE/p1728375309571339?thread_ts=1728048850.634099&cid=C0760NPT2BE

File: src/utils/dateHelpers.test.ts [hunk 4]
```diff
@@ -1,138 +1,161 @@
...
+      {
+        props: ['2023-07-07T00:00:00.000Z'],
+        expected: { isValid: true, date: '2023-07-07T00:00:00.000Z', input: undefined },
+      },
+      {
+        props: ['2023-12-31T23:00:00.000Z'],
+        expected: { isValid: true, date: '2023-12-31T23:00:00.000Z', input: undefined },
+      },
+    ];
+    tests.forEach(({ props, expected }) => {
```
[the remark this answers is not in the document]
Comment (paal2707) [reply src/utils/dateHelpers.test.ts:130]: Fikset, it.each er bedre! :)

**Pull request merged**